cmdstanpy/utils.py

```diff
@@ -216,6 +216,56 @@ def rdump(path: str, data: Dict) -> None:
             fd.write('\n')
 
 
+def rload(fname: str) -> dict:
+    """Parse data and parameter variable values from an R dump format file.
+       This parser only supports the subset of R dump data as described
+       in the "Dump Data Format" section of the CmdStan manual, i.e.,
+       scalar, vector, matrix, and array data types.
+    """
+    data_dict = {}
+    with open(fname, 'r') as fp:
+        lines = fp.readlines()
+
+    # Variable data may span multiple lines, parse accordingly
+    idx = 0
+    while idx < len(lines) and '<-' not in lines[idx]:
+        idx += 1
+    if idx == len(lines):
+        return None
+    start_idx = idx
+    idx += 1
+    while True:
+        while idx < len(lines) and '<-' not in lines[idx]:
+            idx += 1
+        next_var = idx
+        var_data = ''.join(lines[start_idx:next_var]).replace('\n', '')
+        lhs, rhs = [_.strip() for _ in var_data.split('<-')]
+        lhs = lhs.replace('"','')
+        rhs = rhs.replace('L','')
+        if rhs.startswith('structure'):
+            *_, vals, dim = rhs.replace('(', ' ').replace(')', ' ').split('c')
+            vals = [float(v) for v in vals.split(',')[:-1]]
+            dim = [int(v) for v in dim.split(',')]
+            val = np.array(vals).reshape(dim[::-1]).T
+        elif rhs.startswith('c'):
+            val = np.array([float(_) for _ in rhs[2:-1].split(',')])
+        else:
+            try:
+                val = int(rhs)
+            except TypeError:
+                try:
+                    val = float(rhs)
+                except TypeError:
+                    raise ValueError(rhs)
+        data_dict[lhs] = val
+        if idx == len(lines):
```

Comment:
should this check be after `idx += 1` ?

Reply:
no, this logic is correct.
line 238  will have done the incrementing already.
the loop is set up  this way to make sure that the final var decl gets processed.